src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutSignatureNode.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Internal.Text;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a native layout signature. A signature is a pair where the first item is a pointer
+    /// to the TypeManager that contains the native layout info blob of interest, and the second item
+    /// is an offset into that native layout info blob
+    /// </summary>
+    class NativeLayoutSignatureNode : ObjectNode, ISymbolNode
+    {
+        private static int s_counter = 0;
```

Comment:
We need a name mangling scheme - this won't work.

Static field will be a pain to deal with when we're compiling in a multithreaded fashion and the node creation order gets randomized.

@davidwrighton @fadimounir